src/Grapevine.Tests/Server/PublicFolderFacts.cs

```diff
@@ -18,133 +19,209 @@ private static string GenerateUniqueString()
+                folder.CleanUp();
             }
 
             [Fact]
-            public void ThrowsExceptionWhenSetToNull()
+            public void PathAndPrefixSetsPrefix()
             {
-                var root = new PublicFolder();
-                Should.Throw<ArgumentNullException>(() => root.FolderPath = null);
+                var path = Path.Combine(Directory.GetCurrentDirectory(), GenerateUniqueString());
+                const string prefix = "testing";
+
+                var folder = new PublicFolder(path, prefix);
+
+                folder.IndexFileName.ShouldBe(PublicFolder.DefaultIndexFileName);
+                folder.FolderPath.ShouldBe(path);
+                folder.Prefix.Equals($"/{prefix}").ShouldBeTrue();
+                folder.DirectoryListing.Any().ShouldBeFalse();
+
+                folder.CleanUp();
             }
         }
 
-        public class GetFilePathMethod
+        public class DefaultFileNameProperty
         {
             [Fact]
-            public void ReturnsNullWhenPathInfoIsNull()
+            public void DefaultFileNameShowsInDirectoryListingForFileAndFolder()
             {
-                var root = new PublicFolder();
-                root.FilePathGetter(null).ShouldBeNull();
+                var path = Path.Combine(Directory.GetCurrentDirectory(), GenerateUniqueString());
+
+                Directory.CreateDirectory(path);
+                File.WriteAllText(Path.Combine(path, PublicFolder.DefaultIndexFileName), "for testing purposes - delete me");
+
+                var folder = new PublicFolder(path);
+
+                folder.DirectoryListing.Count.ShouldBe(2);
+                folder.DirectoryListing.Count(x => x.Key.EndsWith(PublicFolder.DefaultIndexFileName)).ShouldBe(1);
+                folder.DirectoryListing.Count(x => x.Value == Path.Combine(folder.FolderPath, PublicFolder.DefaultIndexFileName)).ShouldBe(2);
+
+                folder.CleanUp();
             }
 
             [Fact]
-            public void ReturnsNullWhenPathInfoIsEmptyString()
+            public void DefaultFileNameDoesNotChangeWhenSetToEmptyString()
             {
-                var root = new PublicFolder();
-                root.FilePathGetter(string.Empty).ShouldBeNull();
+                var folder = new PublicFolder();
+
+                var defaultFileName = folder.IndexFileName;
+                folder.IndexFileName = string.Empty;
+
+                folder.IndexFileName.ShouldBe(defaultFileName);
             }
 
             [Fact]
-            public void ReturnsNullWhenPathInfoIsWhiteSpace()
+            public void DefaultFileNameDoesNotChangeWhenSetToNull()
             {
-                var root = new PublicFolder();
-                root.FilePathGetter(" ").ShouldBeNull();
+                var folder = new PublicFolder();
+
+                var defaultFileName = folder.IndexFileName;
+                folder.IndexFileName = null;
+
+                folder.IndexFileName.ShouldBe(defaultFileName);
```

Comment:
No, I don't expect the user to ever do it, but (since they can) if they do it can cause all kinds of problem when I re-index the folder structure.

Maybe we should revisit this as a future issue.